packages/apm/src/hubextensions.ts

```diff
@@ -51,18 +51,23 @@ function startTransaction(this: Hub, context: TransactionContext): Transaction {
 function startSpan(this: Hub, context: SpanContext): Transaction | Span {
   /**
    * @deprecated
-   * This is here to make sure we don't break users that relied on calling startSpan to create a transaction
-   * with the transaction poperty set.
+   * TODO: consider removing this in a future release.
+   *
+   * This is for backwards compatibility with releases before startTransaction
+   * existed, to allow for a smoother transition.
    */
-  if ((context as any).transaction !== undefined) {
-    logger.warn(`Use \`Sentry.startTransaction({name: ${(context as any).transaction}})\` to start a Transaction.`);
-    (context as TransactionContext).name = (context as any).transaction as string;
-  }
-
-  // We have the check of not undefined since we defined it's ok to start a transaction with an empty name
-  // tslint:disable-next-line: strict-type-predicates
-  if ((context as TransactionContext).name !== undefined) {
-    return this.startTransaction(context as TransactionContext);
+  {
+    // The `TransactionContext.name` field used to be called `transaction`.
+    const transactionContext = context as Partial<TransactionContext & { transaction: string }>;
+    if (transactionContext.transaction !== undefined) {
+      transactionContext.name = transactionContext.transaction;
+    }
+    // Check for not undefined since we defined it's ok to start a transaction
```

Comment:
So is it ok or not? Because now, we start transaction only if name is defined, yet we say that it can be empty and it's still fine.

Reply:
> So is it ok or not?

It is okay to start without a name and set it later, before calling `Transaction.finish`.

> we start transaction only if name is defined

The comment (which is just being moved/reworded) is highlighting the fact that `if (transactionContext.name)` and `if (transactionContext.name !== undefined)` evaluate differently when name is the empty string.

Also, this is `startSpan` being used to start a transaction as a best effort to support old code of beta users. If old code was starting transactions without a name and such that name was `undefined`, then we cannot make the distinction here, that old code will perhaps erroneously create a Span now.

The behavior is not changing in this PR, at least not intentionally.